Index: linux-3.3.8/arch/mips/ath79/Kconfig
===================================================================
--- linux-3.3.8.orig/arch/mips/ath79/Kconfig	2014-11-11 17:49:21.531115263 +0100
+++ linux-3.3.8/arch/mips/ath79/Kconfig	2014-11-11 18:00:19.415111081 +0100
@@ -431,6 +431,16 @@
 	select ATH79_DEV_LEDS_GPIO
 	select ATH79_DEV_M25P80
 
+config ATH79_MACH_GL_INET
+	bool "GL-INET support"
+	select SOC_AR933X
+	select ATH79_DEV_ETH
+	select ATH79_DEV_GPIO_BUTTONS
+	select ATH79_DEV_LEDS_GPIO
+	select ATH79_DEV_M25P80
+	select ATH79_DEV_USB
+	select ATH79_DEV_WMAC
+
 config ATH79_MACH_TL_MR11U
 	bool "TP-LINK TL-MR11U/TL-MR3040 support"
 	select SOC_AR933X
Index: linux-3.3.8/arch/mips/ath79/Makefile
===================================================================
--- linux-3.3.8.orig/arch/mips/ath79/Makefile	2014-11-11 17:49:21.531115263 +0100
+++ linux-3.3.8/arch/mips/ath79/Makefile	2014-11-11 18:02:48.983111781 +0100
@@ -74,6 +74,7 @@
 obj-$(CONFIG_ATH79_MACH_TEW_632BRP)	+= mach-tew-632brp.o
 obj-$(CONFIG_ATH79_MACH_TEW_673GRU)	+= mach-tew-673gru.o
 obj-$(CONFIG_ATH79_MACH_TEW_712BR)	+= mach-tew-712br.o
+obj-$(CONFIG_ATH79_MACH_GL_INET)	+= mach-gl-inet.o
 obj-$(CONFIG_ATH79_MACH_TL_MR11U)	+= mach-tl-mr11u.o
 obj-$(CONFIG_ATH79_MACH_TL_MR13U)       += mach-tl-mr13u.o
 obj-$(CONFIG_ATH79_MACH_TL_MR3020)	+= mach-tl-mr3020.o
Index: linux-3.3.8/arch/mips/ath79/machtypes.h
===================================================================
--- linux-3.3.8.orig/arch/mips/ath79/machtypes.h	2014-11-11 17:49:21.679115263 +0100
+++ linux-3.3.8/arch/mips/ath79/machtypes.h	2014-11-11 18:01:19.711117105 +0100
@@ -71,6 +71,7 @@
 	ATH79_MACH_TEW_632BRP,		/* TRENDnet TEW-632BRP */
 	ATH79_MACH_TEW_673GRU,		/* TRENDnet TEW-673GRU */
 	ATH79_MACH_TEW_712BR,		/* TRENDnet TEW-712BR */
+	ATH79_MACH_GL_INET,		/* GL-CONNECT GL-INET */
 	ATH79_MACH_TL_MR10U,		/* TP-LINK TL-MR10U */
 	ATH79_MACH_TL_MR11U,		/* TP-LINK TL-MR11U */
 	ATH79_MACH_TL_MR13U,		/* TP-LINK TL-MR13U */
